mod-source-record-storage-server/src/main/java/org/folio/services/SnapshotService.java

```diff
@@ -62,4 +62,14 @@ Future<SnapshotCollection> getSnapshots(Condition condition, Collection<OrderFie
    */
   Future<Boolean> deleteSnapshot(String id, String tenantId);
 
-}
\ No newline at end of file
+  /**
+   * Copy {@link Snapshot} to other tenant
+   *
+   * @param snapshotId       Snapshot id
+   * @param sourceTenantId source tenant id
+   * @param targetTenantId target tenant id
+   * @return future with copied Snapshot
+   */
+  Future<Snapshot> copySnapshotToOtherTenant(String snapshotId, String sourceTenantId, String targetTenantId);
```

Comment:
Please educate me about snapshots, they are really "job executions" from SRM correct?

Updating a central tenant instance means that it already exists against the original snapshot that created it. Updating the central tenant instance means adding a new generation for the record right? if so, do we need to declare a new snapshot?
Thinking the current code will create snapshots in SRS that don't have corresponding job executions in SRM.

Reply:
Yes, you are right. 
So the main thing of this logic - is that we can't manipulate the record (in this way save record) if we don't have a snapshot for it. And in our consortium scope - if we want to save a record on the central tenant - it wont exist there (because there is NO data-import job there, it will be only on the member-tenant where di-job was executed). We discussed with back-end team and decided to just copy the snapshot from the member tenant to the central one, if there is consortium manipulations. So we won't create jobExecution/job info or smth else, this will be just kind of a stub snapshot with stub data to be able to manipulate records in the scope of the central tenant. Feel free to ask any questions.

Reply:
Is a snapshot needed to create another generation of the source record?

Reply:
@okolawole-ebsco  As far as I can see here in scope of code there are no operations regarding snapshots while updating record (creating another generation). But it can assumed that it is already in the record.

Reply:
@okolawole-ebsco  It seems like there is the case when it won't be exits(I tested it already): when we create an Instance manually (not via data-import) on the member-tenant. After that, we will share it to the central one. And trying to update it via data-import on the central tenant. So the source of the Instance will be = FOLIO(after manually creation) but after mapping inside di-core it will be MARC (during data-import job). And this flow will look into srs to retrieve MARC-record (because source=MARC not FOLIO), and cant find this MARC-record on the central tenant. That's why there will be created brand new MARC-record in this case.